openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py

```diff
@@ -352,12 +285,18 @@ def build_dag(
 
     with DAG(**build_dag_configs(ingestion_pipeline)) as dag:
 
+        # Initialize with random UUID4. Will be used by the callback instead of
+        # generating it inside the Workflow itself.
+        workflow_config.pipelineRunId = str(uuid.uuid4())
+
         PythonOperator(
             task_id=task_name,
             python_callable=workflow_fn,
             op_kwargs={"workflow_config": workflow_config},
-            retries=ingestion_pipeline.airflowConfig.retries,
-            retry_delay=ingestion_pipeline.airflowConfig.retryDelay,
+            # There's no need to retry if we have had an error. Wait until the next schedule or manual rerun.
+            retries=0,
+            # each DAG will call its own OpenMetadataWorkflowConfig
+            on_failure_callback=partial(send_failed_status_callback, workflow_config),
```

Comment:
partial func allows us to pass DAG-specific params such as the workflow config. Otherwise only the `context` is available in callbacks